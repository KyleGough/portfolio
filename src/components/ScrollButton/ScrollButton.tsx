import React, { MouseEventHandler } from 'react';

interface ScrollButtonProps {
  onClick: MouseEventHandler<HTMLButtonElement>;
  children: React.ReactNode;
}

export const ScrollButton: React.FC<ScrollButtonProps> = ({
  onClick,
  children,
}) => (
  <button
    onClick={onClick}
    className="shimmer max-w-[22rem] md:h-[78px] mx-auto w-full px-12 py-4 bg-background text-link hover:text-link-hover focus:text-link-hover rounded-sm shadow border-light"
  >
    {children}
  </button>
);
