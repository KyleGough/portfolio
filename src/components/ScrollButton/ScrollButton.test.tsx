import { render, screen } from '@testing-library/react';
import React from 'react';

import { ScrollButton } from './ScrollButton';

describe('ScrollButton component', () => {
  it('renders', async () => {
    const { container } = render(
      <ScrollButton onClick={() => jest.fn()}>Test</ScrollButton>
    );
    expect(screen.getByText('Test')).toBeVisible();
    expect(container).toMatchSnapshot();
  });
});
