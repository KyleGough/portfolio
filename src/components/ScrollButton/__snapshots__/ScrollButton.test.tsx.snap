// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollButton component renders 1`] = `
<div>
  <button
    class="shimmer max-w-[22rem] md:h-[78px] mx-auto w-full px-12 py-4 bg-background text-link hover:text-link-hover focus:text-link-hover rounded-sm shadow border-light"
  >
    Test
  </button>
</div>
`;
